From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Fri, 16 May 2014 11:29:22 +0200
Subject: libs/web: remove strange 'static' on variable declaration

diff --git a/modules/base/src/template_parser.c b/modules/base/src/template_parser.c
index fc8607b..1aa5131 100644
--- a/modules/base/src/template_parser.c
+++ b/modules/base/src/template_parser.c
@@ -66,7 +66,7 @@ static char *strfind(char *haystack, int hslen, const char *needle, int ndlen)
 struct template_parser * template_open(const char *file)
 {
 	struct stat s;
-	static struct template_parser *parser;
+	struct template_parser *parser;
 
 	if (!(parser = malloc(sizeof(*parser))))
 		goto err;
