From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Sat, 26 Jul 2014 18:06:07 +0200
Subject: haveged: start directly after initial setup and don't run tests

diff --git a/utils/haveged/files/haveged.init b/utils/haveged/files/haveged.init
index ce28e61..aa8d3c7 100644
--- a/utils/haveged/files/haveged.init
+++ b/utils/haveged/files/haveged.init
@@ -1,14 +1,16 @@
 #!/bin/sh /etc/rc.common
 # Copyright (C) 2012 OpenWrt.org
 
-START=98
+START=13
 
 HAVEGED_THRESHOLD=1024
 HAVEGED_DCACHE=32
 HAVEGED_ICACHE=32
+HAVEGED_TESTS_TOT=
+HAVEGED_TESTS_CONT=
 
 start() {
-	service_start /usr/sbin/haveged -w $HAVEGED_THRESHOLD -d $HAVEGED_DCACHE -i $HAVEGED_ICACHE -v 1
+	service_start /usr/sbin/haveged -w $HAVEGED_THRESHOLD -d $HAVEGED_DCACHE -i $HAVEGED_ICACHE -v 1 -o t${HAVEGED_TESTS_TOT}c${HAVEGED_TESTS_CONT}
 }
 
 stop() {
