From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Wed, 28 Oct 2015 01:10:09 +0100
Subject: node: add python/host to HOST_BUILD_DEPENDS as well

Signed-off-by: Matthias Schiffer <mschiffer@universe-factory.net>

diff --git a/lang/node/Makefile b/lang/node/Makefile
index 65a5390..243c8a5 100644
--- a/lang/node/Makefile
+++ b/lang/node/Makefile
@@ -14,6 +14,7 @@ PKG_RELEASE:=1
 PKG_SOURCE:=node-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=http://nodejs.org/dist/${PKG_VERSION}
 
+HOST_BUILD_DEPENDS:=python/host
 PKG_BUILD_DEPENDS:=python/host
 PKG_INSTALL:=1
 PKG_USE_MIPS16:=0
