From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Sat, 26 Jul 2014 18:06:07 +0200
Subject: haveged: start directly after initial setup and disable tests

diff --git a/utils/haveged/Makefile b/utils/haveged/Makefile
index 12a6237..e9ff3f1 100644
--- a/utils/haveged/Makefile
+++ b/utils/haveged/Makefile
@@ -41,8 +41,9 @@ define Package/libhavege
 endef
 
 CONFIGURE_ARGS+= \
-      --enable-daemon=yes
-      --enable-threads=no
+      --enable-daemon=yes \
+      --enable-threads=no \
+      --enable-olt=no
 
 define Build/InstallDev
 	$(INSTALL_DIR) $(1)/usr/include
diff --git a/utils/haveged/files/haveged.init b/utils/haveged/files/haveged.init
index ce28e61..113d64f 100644
--- a/utils/haveged/files/haveged.init
+++ b/utils/haveged/files/haveged.init
@@ -1,7 +1,7 @@
 #!/bin/sh /etc/rc.common
 # Copyright (C) 2012 OpenWrt.org
 
-START=98
+START=13
 
 HAVEGED_THRESHOLD=1024
 HAVEGED_DCACHE=32
