From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Sun, 12 Jan 2014 00:44:29 +0100
Subject: haveged: start directly after the root partition has been mounted

diff --git a/utils/haveged/files/haveged.init b/utils/haveged/files/haveged.init
index 129d951..2c5be1d 100644
--- a/utils/haveged/files/haveged.init
+++ b/utils/haveged/files/haveged.init
@@ -1,7 +1,7 @@
 #!/bin/sh /etc/rc.common
 # Copyright (C) 2012 OpenWrt.org
 
-START=98
+START=11
 
 HAVEGED_THRESHOLD=1024
 HAVEGED_DCACHE=32
