From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Wed, 30 Dec 2015 01:00:49 +0100
Subject: lua-cjson: add host build support

diff --git a/lang/lua-cjson/Makefile b/lang/lua-cjson/Makefile
index fd489f2..90c8f98 100644
--- a/lang/lua-cjson/Makefile
+++ b/lang/lua-cjson/Makefile
@@ -20,6 +20,7 @@ PKG_MD5SUM:=24f270663e9f6ca8ba2a02cef19f7963
 
 PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)
 
+include $(INCLUDE_DIR)/host-build.mk
 include $(INCLUDE_DIR)/package.mk
 include $(INCLUDE_DIR)/cmake.mk
 
@@ -39,6 +40,9 @@ endef
 CMAKE_OPTIONS += \
 	-DUSE_LUA=ON
 
+CMAKE_HOST_OPTIONS += \
+	-DLUA_MATH_LIBRARY=m
+
 define Package/lua-cjson/install
 	$(INSTALL_DIR) $(1)/usr/lib/lua
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/cjson.so $(1)/usr/lib/lua/
@@ -47,4 +51,5 @@ define Package/lua-cjson/install
 	$(INSTALL_DATA) $(PKG_BUILD_DIR)/lua/cjson/util.lua $(1)/usr/lib/lua/cjson
 endef
 
+$(eval $(call HostBuild))
 $(eval $(call BuildPackage,lua-cjson))
