From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Tue, 29 Mar 2016 00:24:00 +0200
Subject: fastd: update to v18

Signed-off-by: Matthias Schiffer <mschiffer@universe-factory.net>

diff --git a/net/fastd/Config.in b/net/fastd/Config.in
index 5820b9796562cd11826f8bbefbb02806fe516c61..3350eb3099a26c870d70373c0712a8b59881ee5c 100644
--- a/net/fastd/Config.in
+++ b/net/fastd/Config.in
@@ -74,6 +74,12 @@ config FASTD_ENABLE_MAC_UHASH
 	default y
 
 
+config FASTD_WITH_CAPABILITIES
+	bool "Enable POSIX capability support"
+	depends on PACKAGE_fastd
+	default n
+
+
 config FASTD_WITH_CMDLINE_USER
 	bool "Include support for setting user/group related options on the command line"
 	depends on PACKAGE_fastd
diff --git a/net/fastd/Makefile b/net/fastd/Makefile
index 027f5491d2f6b066affe08e846b1967900c391e1..5c550d8a5f4669d4f018dde23b622376bc8d0b06 100644
--- a/net/fastd/Makefile
+++ b/net/fastd/Makefile
@@ -8,13 +8,13 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=fastd
-PKG_VERSION:=17
+PKG_VERSION:=18
 PKG_RELEASE:=1
 
 PKG_MAINTAINER:=Matthias Schiffer <mschiffer@universe-factory.net>
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
-PKG_SOURCE_URL:=https://projects.universe-factory.net/attachments/download/81
-PKG_MD5SUM:=bad4f1948702f418b799578f83a0edb8
+PKG_SOURCE_URL:=https://projects.universe-factory.net/attachments/download/86
+PKG_MD5SUM:=e53236d3049f64f7955ad9556da099eb
 
 PKG_LICENSE:=BSD-2-Clause
 PKG_LICENSE_FILES:=COPYRIGHT
@@ -34,6 +34,7 @@ PKG_CONFIG_DEPENDS:=\
 	CONFIG_FASTD_ENABLE_CIPHER_SALSA2012 \
 	CONFIG_FASTD_ENABLE_MAC_GHASH \
 	CONFIG_FASTD_ENABLE_MAC_UHASH \
+	CONFIG_FASTD_WITH_CAPABILITIES \
 	CONFIG_FASTD_WITH_CMDLINE_USER \
 	CONFIG_FASTD_WITH_CMDLINE_LOGGING \
 	CONFIG_FASTD_WITH_CMDLINE_OPERATION \
@@ -50,7 +51,7 @@ include $(INCLUDE_DIR)/cmake.mk
 define Package/fastd
   SECTION:=net
   CATEGORY:=Network
-  DEPENDS:=+kmod-tun +librt +libpthread +FASTD_WITH_STATUS_SOCKET:libjson-c
+  DEPENDS:=+kmod-tun +librt +libpthread +FASTD_WITH_STATUS_SOCKET:libjson-c +FASTD_WITH_CAPABILITIES:libcap
   TITLE:=Fast and Secure Tunneling Daemon
   URL:=https://projects.universe-factory.net/projects/fastd
   SUBMENU:=VPN
@@ -79,13 +80,13 @@ CMAKE_OPTIONS += \
 	-DWITH_CIPHER_SALSA2012:BOOL=FALSE \
 	-DWITH_MAC_GHASH:BOOL=FALSE \
 	-DWITH_MAC_UHASH:BOOL=FALSE \
+	-DWITH_CAPABILITIES:BOOL=FALSE \
 	-DWITH_CMDLINE_USER:BOOL=FALSE \
 	-DWITH_CMDLINE_LOGGING:BOOL=FALSE \
 	-DWITH_CMDLINE_OPERATION:BOOL=FALSE \
 	-DWITH_CMDLINE_COMMANDS:BOOL=FALSE \
 	-DWITH_DYNAMIC_PEERS:BOOL=FALSE \
 	-DWITH_STATUS_SOCKET:BOOL=FALSE \
-	-DWITH_CAPABILITIES:BOOL=FALSE \
 	-DENABLE_SYSTEMD:BOOL=FALSE \
 	-DENABLE_LIBSODIUM:BOOL=FALSE \
 	-DENABLE_LTO:BOOL=TRUE
@@ -150,6 +151,11 @@ CMAKE_OPTIONS += -DWITH_MAC_UHASH:BOOL=TRUE
 endif
 
 
+ifeq ($(CONFIG_FASTD_WITH_CAPABILITIES),y)
+CMAKE_OPTIONS += -DWITH_CAPABILITIES:BOOL=TRUE
+endif
+
+
 ifeq ($(CONFIG_FASTD_WITH_CMDLINE_USER),y)
 CMAKE_OPTIONS += -DWITH_CMDLINE_USER:BOOL=TRUE
 endif
