From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Mon, 9 May 2016 15:57:18 +0200
Subject: node: update to v0.12.14

While we're at it, also enable parallel builds.

Signed-off-by: Matthias Schiffer <mschiffer@universe-factory.net>

diff --git a/lang/node/Makefile b/lang/node/Makefile
index 243c8a5..ed35e17 100644
--- a/lang/node/Makefile
+++ b/lang/node/Makefile
@@ -8,17 +8,21 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=node
-PKG_VERSION:=v0.12.7
+PKG_VERSION:=v0.12.14
 PKG_RELEASE:=1
 
-PKG_SOURCE:=node-$(PKG_VERSION).tar.gz
+PKG_SOURCE:=node-$(PKG_VERSION).tar.xz
 PKG_SOURCE_URL:=http://nodejs.org/dist/${PKG_VERSION}
+PKG_MD5SUM:=27f1a2cf00af32cbfe9401ca4b1a805f
 
 HOST_BUILD_DEPENDS:=python/host
 PKG_BUILD_DEPENDS:=python/host
 PKG_INSTALL:=1
 PKG_USE_MIPS16:=0
 
+HOST_BUILD_PARALLEL:=1
+PKG_BUILD_PARALLEL:=1
+
 PKG_MAINTAINER:=John Crispin <blogic@openwrt.org>
 PKG_LICENSE:=
 
@@ -56,6 +60,12 @@ HOST_CONFIGURE_ARGS:= \
 
 HOST_CONFIGURE_CMD:=python ./configure
 
+HOST_MAKE_FLAGS += CXXFLAGS='-std=c++11'
+
+define Host/Install
+	$(MAKE) -C $(HOST_BUILD_DIR) $(HOST_MAKE_FLAGS) install
+endef
+
 define Build/InstallDev
 	$(INSTALL_DIR) $(1)/usr/include
 	$(CP) $(PKG_INSTALL_DIR)/usr/include/* $(1)/usr/include/
