From: Stijn Tintel <stijn@linux-ipv6.be>
Date: Thu, 17 Mar 2016 10:35:53 +0100
Subject: node-cylon: add node host dependency

Signed-off-by: Stijn Tintel <stijn@linux-ipv6.be>

diff --git a/lang/node-cylon/Makefile b/lang/node-cylon/Makefile
index c2e3b7265d11316dabd22dbf0f6ea8ee83fdeb5a..d2ed0a0a25c65e8b13536dc1c44b498743766ef7 100644
--- a/lang/node-cylon/Makefile
+++ b/lang/node-cylon/Makefile
@@ -18,7 +18,7 @@ PKG_SOURCE_VERSION:=0c37da77e48b3e2cc3a8d566822a17689de91b40
 PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_SOURCE_VERSION)
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_SOURCE_VERSION).tar.gz
 
-PKG_BUILD_DEPENDS:=node
+PKG_BUILD_DEPENDS:=node/host
 PKG_NODE_VERSION:=0.12.7
 
 PKG_MAINTAINER:=John Crispin <blogic@openwrt.org>
