From: Hirokazu MORIKAWA <register@nxhack.com>
Date: Fri, 1 Apr 2016 13:46:31 +0900
Subject: node-* : specify npm_config_cache directory

diff --git a/lang/node-arduino-firmata/Makefile b/lang/node-arduino-firmata/Makefile
index 96d120f5698a783e54384b05c32d11ed9b1bd319..89e9637be980fff6c216ba62d1bd617d63e2e130 100644
--- a/lang/node-arduino-firmata/Makefile
+++ b/lang/node-arduino-firmata/Makefile
@@ -52,6 +52,7 @@ define Build/Compile
 	$(MAKE_FLAGS) \
 	npm_config_arch=$(CONFIG_ARCH) \
 	npm_config_nodedir=$(BUILD_DIR)/node-v$(PKG_NODE_VERSION)/ \
+	npm_config_cache=$(BUILD_DIR)/node-v$(PKG_NODE_VERSION)/npm-cache \
 	PREFIX="$(PKG_INSTALL_DIR)/usr/" \
 	$(STAGING_DIR_HOST)/bin/npm install -g $(PKG_BUILD_DIR)
 endef
