From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Wed, 7 May 2014 22:57:10 +0200
Subject: tools/Makefile: fix host tools build dependency

diff --git a/tools/Makefile b/tools/Makefile
index 2236373..88c0562 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -100,7 +100,7 @@ $(STAGING_DIR_HOST)/.prepared: $(TMP_DIR)/.build
 
 
 define PrepareCommand
-$(STAGING_DIR_HOST)/bin/$(1): $(STAGING_DIR)/.prepared
+$(STAGING_DIR_HOST)/bin/$(1): $(STAGING_DIR_HOST)/.prepared
 	@mkdir -p "$$(dir $$@)"; rm -f "$$@"
 	@export FILE="$$$$(which $(2) 2>/dev/null | grep -v 'not found' | head -n1)"; [ -n "$$$$FILE" ] || { \
 		echo "Command $(1) not found."; false; \
@@ -109,7 +109,7 @@ $(STAGING_DIR_HOST)/bin/$(1): $(STAGING_DIR)/.prepared
 endef
 endif
 
-$(STAGING_DIR_HOST)/bin/stat: $(STAGING_DIR)/.prepared
+$(STAGING_DIR_HOST)/bin/stat: $(STAGING_DIR_HOST)/.prepared
 	@rm -f $@
 	@if stat --version > /dev/null 2>&1; then \
 		ln -s `which stat` $@; \
