From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Sat, 16 Aug 2014 17:52:34 +0200
Subject: ar71xx: correctly detect hardware revision on TP-Link Archer C5 and C7

diff --git a/target/linux/ar71xx/base-files/lib/ar71xx.sh b/target/linux/ar71xx/base-files/lib/ar71xx.sh
index 18da356..1709356 100755
--- a/target/linux/ar71xx/base-files/lib/ar71xx.sh
+++ b/target/linux/ar71xx/base-files/lib/ar71xx.sh
@@ -214,6 +214,13 @@ tplink_board_detect() {
 	"934100"*)
 		model="NC-LINK SMART-300"
 		;;
+	"c50000"*)
+		model="TP-Link Archer C5"
+		;;
+	"750000"*|\
+	"c70000"*)
+		model="TP-Link Archer C7"
+		;;
 	*)
 		hwver=""
 		;;
@@ -748,7 +755,7 @@ ar71xx_board_detect() {
 	esac
 
 	case "$machine" in
-	*TL-WR* | *TL-WA* | *TL-MR* | *TL-WD*)
+	*TL-WR* | *TL-WA* | *TL-MR* | *TL-WD* | *Archer*)
 		tplink_board_detect "$machine"
 		;;
 	esac
