From: David Bauer <mail@david-bauer.net>
Date: Sun, 5 Jun 2022 23:43:38 +0200
Subject: ramips-mt7621: make DSA images swconfig upgradable

diff --git a/target/linux/ramips/image/mt7621.mk b/target/linux/ramips/image/mt7621.mk
index 961f8a3b7514a3658f8d4795c136d5069830126f..89329729b1c035642311363e8aad46ba5b8d4d22 100644
--- a/target/linux/ramips/image/mt7621.mk
+++ b/target/linux/ramips/image/mt7621.mk
@@ -179,7 +179,6 @@ endef
 TARGET_DEVICES += asiarf_ap7621-nv1
 
 define Device/asus_rt-ac57u
-  $(Device/dsa-migration)
   DEVICE_VENDOR := ASUS
   DEVICE_MODEL := RT-AC57U
   DEVICE_ALT0_VENDOR := ASUS
@@ -426,7 +425,6 @@ endef
 TARGET_DEVICES += dlink_dir-853-r1
 
 define Device/dlink_dir-860l-b1
-  $(Device/dsa-migration)
   $(Device/seama-lzma-loader)
   SEAMA_SIGNATURE := wrgac13_dlink.2013gui_dir860lb
   IMAGE_SIZE := 16064k
@@ -1233,7 +1231,6 @@ endef
 TARGET_DEVICES += mts_wg430223
 
 define Device/netgear_ex6150
-  $(Device/dsa-migration)
   $(Device/uimage-lzma-loader)
   DEVICE_VENDOR := NETGEAR
   DEVICE_MODEL := EX6150
@@ -1246,7 +1243,6 @@ endef
 TARGET_DEVICES += netgear_ex6150
 
 define Device/netgear_sercomm_nand
-  $(Device/dsa-migration)
   $(Device/uimage-lzma-loader)
   BLOCKSIZE := 128k
   PAGESIZE := 2048
@@ -1429,7 +1425,6 @@ endef
 TARGET_DEVICES += netgear_wax202
 
 define Device/netgear_wndr3700-v5
-  $(Device/dsa-migration)
   $(Device/netgear_sercomm_nor)
   $(Device/uimage-lzma-loader)
   IMAGE_SIZE := 15232k
@@ -1754,7 +1749,6 @@ endef
 TARGET_DEVICES += tplink_tl-wpa8631p-v3
 
 define Device/ubnt_edgerouter_common
-  $(Device/dsa-migration)
   $(Device/uimage-lzma-loader)
   DEVICE_VENDOR := Ubiquiti
   IMAGE_SIZE := 256768k
@@ -2170,7 +2164,6 @@ endef
 TARGET_DEVICES += zbtlink_zbt-wg2626
 
 define Device/zbtlink_zbt-wg3526-16m
-  $(Device/dsa-migration)
   $(Device/uimage-lzma-loader)
   IMAGE_SIZE := 16064k
   DEVICE_VENDOR := Zbtlink
@@ -2183,7 +2176,6 @@ endef
 TARGET_DEVICES += zbtlink_zbt-wg3526-16m
 
 define Device/zbtlink_zbt-wg3526-32m
-  $(Device/dsa-migration)
   $(Device/uimage-lzma-loader)
   IMAGE_SIZE := 32448k
   DEVICE_VENDOR := Zbtlink
