From: David Bauer <mail@david-bauer.net>
Date: Sun, 5 Jun 2022 23:43:38 +0200
Subject: ramips-mt7621: make DSA images swconfig upgradable

diff --git a/target/linux/ramips/image/mt7621.mk b/target/linux/ramips/image/mt7621.mk
index 8860be11b8c879c067901cf6642a133b27b1e3b2..f8a846cfc67c306fedfbdb2bd2056fa3d6aa1b61 100644
--- a/target/linux/ramips/image/mt7621.mk
+++ b/target/linux/ramips/image/mt7621.mk
@@ -172,7 +172,6 @@ endef
 TARGET_DEVICES += asiarf_ap7621-nv1
 
 define Device/asus_rt-ac57u
-  $(Device/dsa-migration)
   DEVICE_VENDOR := ASUS
   DEVICE_MODEL := RT-AC57U
   DEVICE_ALT0_VENDOR := ASUS
@@ -418,7 +417,6 @@ endef
 TARGET_DEVICES += dlink_dir-853-r1
 
 define Device/dlink_dir-860l-b1
-  $(Device/dsa-migration)
   $(Device/seama)
   SEAMA_SIGNATURE := wrgac13_dlink.2013gui_dir860lb
   LOADER_TYPE := bin
@@ -1226,7 +1224,6 @@ endef
 TARGET_DEVICES += mts_wg430223
 
 define Device/netgear_ex6150
-  $(Device/dsa-migration)
   DEVICE_VENDOR := NETGEAR
   DEVICE_MODEL := EX6150
   DEVICE_PACKAGES := kmod-mt76x2
@@ -1238,7 +1235,6 @@ endef
 TARGET_DEVICES += netgear_ex6150
 
 define Device/netgear_sercomm_nand
-  $(Device/dsa-migration)
   $(Device/uimage-lzma-loader)
   BLOCKSIZE := 128k
   PAGESIZE := 2048
@@ -1421,7 +1417,6 @@ endef
 TARGET_DEVICES += netgear_wax202
 
 define Device/netgear_wndr3700-v5
-  $(Device/dsa-migration)
   $(Device/netgear_sercomm_nor)
   $(Device/uimage-lzma-loader)
   IMAGE_SIZE := 15232k
@@ -1745,7 +1740,6 @@ endef
 TARGET_DEVICES += tplink_tl-wpa8631p-v3
 
 define Device/ubnt_edgerouter_common
-  $(Device/dsa-migration)
   $(Device/uimage-lzma-loader)
   DEVICE_VENDOR := Ubiquiti
   IMAGE_SIZE := 256768k
@@ -2142,7 +2136,6 @@ endef
 TARGET_DEVICES += zbtlink_zbt-wg2626
 
 define Device/zbtlink_zbt-wg3526-16m
-  $(Device/dsa-migration)
   $(Device/uimage-lzma-loader)
   IMAGE_SIZE := 16064k
   DEVICE_VENDOR := Zbtlink
@@ -2155,7 +2148,6 @@ endef
 TARGET_DEVICES += zbtlink_zbt-wg3526-16m
 
 define Device/zbtlink_zbt-wg3526-32m
-  $(Device/dsa-migration)
   $(Device/uimage-lzma-loader)
   IMAGE_SIZE := 32448k
   DEVICE_VENDOR := Zbtlink
