From: David Bauer <mail@david-bauer.net>
Date: Tue, 5 Jul 2022 23:49:31 +0200
Subject: lantiq-xrx200: make DSA images swconfig upgradable

diff --git a/target/linux/lantiq/image/tp-link.mk b/target/linux/lantiq/image/tp-link.mk
index 751f61589de8cfa51f75ffe0a151b22e87b2e9dd..3c2a58dee81a8972e0056e9cb5a3824637b34af2 100644
--- a/target/linux/lantiq/image/tp-link.mk
+++ b/target/linux/lantiq/image/tp-link.mk
@@ -1,7 +1,7 @@
 DEVICE_VARS += TPLINK_FLASHLAYOUT TPLINK_HWID TPLINK_HWREV TPLINK_HWREVADD TPLINK_HVERSION
 
 define Device/dsa-migration
-  DEVICE_COMPAT_VERSION := 1.1
+  DEVICE_COMPAT_VERSION := 1.0
   DEVICE_COMPAT_MESSAGE := Config cannot be migrated from swconfig to DSA
 endef
 
diff --git a/target/linux/lantiq/image/vr9.mk b/target/linux/lantiq/image/vr9.mk
index f3149bf2c0789e659d287a62de03fe0a891d713b..f413ca8082e403dd58e794b9e21d27e6cde91aa8 100644
--- a/target/linux/lantiq/image/vr9.mk
+++ b/target/linux/lantiq/image/vr9.mk
@@ -1,7 +1,7 @@
 DEVICE_VARS += NETGEAR_BOARD_ID NETGEAR_HW_ID
 
 define Device/dsa-migration
-  DEVICE_COMPAT_VERSION := 1.1
+  DEVICE_COMPAT_VERSION := 1.0
   DEVICE_COMPAT_MESSAGE := Config cannot be migrated from swconfig to DSA
 endef
 
