From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Sat, 14 Jul 2018 01:17:16 +0200
Subject: mt76: depend on kernel 4.14

Do not break the build on kernel 4.9.

Signed-off-by: Matthias Schiffer <mschiffer@universe-factory.net>

diff --git a/package/kernel/mt76/Makefile b/package/kernel/mt76/Makefile
index 2906849cf1e43424daa090026123737452dd1a61..e53dcb21515495181959ad1f19cd4d21a4e97e07 100644
--- a/package/kernel/mt76/Makefile
+++ b/package/kernel/mt76/Makefile
@@ -23,7 +23,7 @@ include $(INCLUDE_DIR)/package.mk
 define KernelPackage/mt76-default
   SUBMENU:=Wireless Drivers
   DEPENDS:= \
-	+kmod-mac80211 @PCI_SUPPORT @!LINUX_3_18 \
+	+kmod-mac80211 @PCI_SUPPORT @LINUX_4_14 \
 	+@DRIVER_11AC_SUPPORT +@DRIVER_11N_SUPPORT +@DRIVER_11W_SUPPORT
 endef
 
