From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Sun, 21 Jun 2015 05:47:54 +0200
Subject: Revert "6to4: Remove sourcerouting parameter registration"

This reverts commit b64e2fa3f75b9ae21d670a9084ca855ea39c25e6.

diff --git a/package/network/ipv6/6to4/files/6to4.sh b/package/network/ipv6/6to4/files/6to4.sh
index a5d0567..57efc52 100755
--- a/package/network/ipv6/6to4/files/6to4.sh
+++ b/package/network/ipv6/6to4/files/6to4.sh
@@ -91,6 +91,7 @@ proto_6to4_init_config() {
 	proto_config_add_int "mtu"
 	proto_config_add_int "ttl"
 	proto_config_add_string "tos"
+	proto_config_add_boolean "sourcerouting"
 }
 
 [ -n "$INCLUDE_ONLY" ] || {
