From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Mon, 26 Jan 2015 03:53:05 +0100
Subject: BB: gmp: use http instead of ftp download (#18805)

Signed-off-by: Felix Fietkau <nbd@openwrt.org>

Backport of r44082

diff --git a/package/libs/gmp/Makefile b/package/libs/gmp/Makefile
index f5839b7..4826e4a 100644
--- a/package/libs/gmp/Makefile
+++ b/package/libs/gmp/Makefile
@@ -13,7 +13,7 @@ PKG_REVISION:=a
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION)$(PKG_REVISION).tar.xz
-PKG_SOURCE_URL:=ftp://ftp.gmplib.org/pub/$(PKG_NAME)/
+PKG_SOURCE_URL:=http://gmplib.org/download/gmp/
 PKG_MD5SUM:=1e6da4e434553d2811437aa42c7f7c76
 
 PKG_BUILD_PARALLEL:=1
