From: jow <jow@3c298f89-4303-0410-b956-a3cf2f4a3e73>
Date: Thu, 17 Jan 2013 15:46:26 +0000
Subject: x86: explicitely pass staging directory to grub-mkimage instead of relying on build time defaults (#12821)

diff --git a/target/linux/x86/image/Makefile b/target/linux/x86/image/Makefile
index 8f0debd..6e2bd54 100644
--- a/target/linux/x86/image/Makefile
+++ b/target/linux/x86/image/Makefile
@@ -102,6 +102,7 @@ ifneq ($(CONFIG_X86_GRUB_IMAGES),)
 	$(INSTALL_DIR) $(KDIR)/root.grub/boot/grub $(KDIR)/grub2
 	$(CP) $(KDIR)/bzImage $(KDIR)/root.grub/boot/vmlinuz
 	grub-mkimage \
+		-d $(STAGING_DIR_HOST)/lib/grub/i386-pc \
 		-o $(KDIR)/grub2/core.img \
 		-O i386-pc \
 		-c ./grub-early.cfg \
@@ -171,6 +172,7 @@ ifdef CONFIG_X86_USE_GRUB2
 	$(INSTALL_DIR) $(KDIR)/root.grub/boot/grub $(KDIR)/grub2
 	$(CP) $(KDIR)/bzImage $(KDIR)/root.grub/boot/vmlinuz
 	grub-mkimage \
+		-d $(STAGING_DIR_HOST)/lib/grub/i386-pc \
 		-o $(KDIR)/grub2/eltorito.img \
 		-O i386-pc \
 		-c ./grub-early.cfg \
