From: nbd <nbd@3c298f89-4303-0410-b956-a3cf2f4a3e73>
Date: Sun, 30 Mar 2014 19:55:39 +0000
Subject: grub2: disable libdevmapper - fix build when it's available

Signed-off-by: Alexander Couzens <lynxis@fe80.eu>

diff --git a/package/grub2/Makefile b/package/grub2/Makefile
index 4edd92b..ed86fff 100644
--- a/package/grub2/Makefile
+++ b/package/grub2/Makefile
@@ -46,7 +46,8 @@ endef
 CONFIGURE_ARGS += \
 	--target=$(REAL_GNU_TARGET_NAME) \
 	--disable-werror \
-	--disable-nls
+	--disable-nls \
+	--disable-device-mapper
 
 HOST_CONFIGURE_ARGS += \
 	--target=$(REAL_GNU_TARGET_NAME) \
