From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Sat, 5 Jul 2014 22:30:34 +0200
Subject: iptables: avoid file conflicts due to unneeded libip6t_*.so in ip6tables package

diff --git a/package/iptables/Makefile b/package/iptables/Makefile
index e36a093..0c7a380 100644
--- a/package/iptables/Makefile
+++ b/package/iptables/Makefile
@@ -406,10 +406,6 @@ define Package/ip6tables/install
 	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/sbin/ip6tables $(1)/usr/sbin/
 	$(LN) ip6tables $(1)/usr/sbin/ip6tables-save
 	$(LN) ip6tables $(1)/usr/sbin/ip6tables-restore
-	$(INSTALL_DIR) $(1)/usr/lib/iptables
-	(cd $(PKG_INSTALL_DIR)/usr/lib/iptables ; \
-		$(CP) libip6t_*.so $(1)/usr/lib/iptables/ \
-	)
 endef
 
 define Package/libiptc/install
