From: Felix Kaechele <felix@kaechele.ca>
Date: Tue, 17 Nov 2015 05:00:02 +0100
Subject: x86/generic: refresh kernel config

Signed-off-by: Felix Kaechele <felix@kaechele.ca>

diff --git a/target/linux/x86/generic/config-default b/target/linux/x86/generic/config-default
index eb4b357..e077ebd 100644
--- a/target/linux/x86/generic/config-default
+++ b/target/linux/x86/generic/config-default
@@ -17,7 +17,6 @@ CONFIG_ACPI_LEGACY_TABLES_LOOKUP=y
 CONFIG_ACPI_PROCESSOR=y
 # CONFIG_ACPI_PROCESSOR_AGGREGATOR is not set
 # CONFIG_ACPI_PROCFS_POWER is not set
-# CONFIG_ACPI_REDUCED_HARDWARE_ONLY is not set
 # CONFIG_ACPI_SBS is not set
 CONFIG_ACPI_THERMAL=y
 CONFIG_ACPI_VIDEO=y
@@ -46,7 +45,6 @@ CONFIG_BACKLIGHT_LCD_SUPPORT=y
 CONFIG_BLK_DEV_SR=y
 # CONFIG_BLK_DEV_SR_VENDOR is not set
 CONFIG_CPU_IDLE_GOV_MENU=y
-# CONFIG_DELL_SMO8800 is not set
 CONFIG_DMA_SHARED_BUFFER=y
 CONFIG_DMI=y
 # CONFIG_DMIID is not set
@@ -120,7 +118,6 @@ CONFIG_INPUT_MOUSEDEV=y
 CONFIG_INPUT_MOUSEDEV_PSAUX=y
 CONFIG_INPUT_MOUSEDEV_SCREEN_X=1024
 CONFIG_INPUT_MOUSEDEV_SCREEN_Y=768
-# CONFIG_INT340X_THERMAL is not set
 CONFIG_INTEL_GTT=y
 CONFIG_INTEL_IDLE=y
 # CONFIG_INTEL_IPS is not set
@@ -173,13 +170,10 @@ CONFIG_PNP_DEBUG_MESSAGES=y
 CONFIG_RAS=y
 # CONFIG_SAMSUNG_Q10 is not set
 CONFIG_SATA_AHCI=y
-# CONFIG_SCx200_ACB is not set
-# CONFIG_SERIAL_8250_FINTEK is not set
 CONFIG_SERIAL_8250_PNP=y
 # CONFIG_THINKPAD_ACPI is not set
 # CONFIG_TOPSTAR_LAPTOP is not set
 # CONFIG_TOSHIBA_BT_RFKILL is not set
-# CONFIG_TOSHIBA_HAPS is not set
 CONFIG_USB=y
 CONFIG_USB_COMMON=y
 CONFIG_USB_EHCI_HCD=y
