From: Sven Eckelmann <sven.eckelmann@open-mesh.com>
Date: Mon, 14 Sep 2015 20:11:43 +0000
Subject: package/uboot-envtools: add OpenMesh MR1750 support

Signed-off-by: Sven Eckelmann <sven.eckelmann@open-mesh.com>

Backport of r46931

Forwarded: https://patchwork.ozlabs.org/patch/624178/

diff --git a/package/boot/uboot-envtools/files/ar71xx b/package/boot/uboot-envtools/files/ar71xx
index ec8541c..ef00f17 100644
--- a/package/boot/uboot-envtools/files/ar71xx
+++ b/package/boot/uboot-envtools/files/ar71xx
@@ -21,6 +21,7 @@ carambola2 | \
 eap300v2 | \
 hornet-ub | \
 hornet-ub-x2 | \
+mr1750 | \
 mr600 | \
 mr600v2 | \
 mr900 | \
