From: Sven Eckelmann <sven.eckelmann@open-mesh.com>
Date: Sat, 9 Apr 2016 10:26:01 +0000
Subject: om-watchdog: add OpenMesh OM5P-AC support

Signed-off-by: Sven Eckelmann <sven.eckelmann@open-mesh.com>

Backport of r49145

Forwarded: https://patchwork.ozlabs.org/patch/624195/

diff --git a/package/kernel/om-watchdog/files/om-watchdog.init b/package/kernel/om-watchdog/files/om-watchdog.init
index c792968..8cec13b 100644
--- a/package/kernel/om-watchdog/files/om-watchdog.init
+++ b/package/kernel/om-watchdog/files/om-watchdog.init
@@ -22,6 +22,9 @@ boot() {
 			"om5p"|"om5p-an")
 				service_start /sbin/om-watchdog 11
 				;;
+			"om5p-ac")
+				service_start /sbin/om-watchdog 17
+				;;
 			"mr600v2")
 				service_start /sbin/om-watchdog 15
 				;;
