From: Sven Eckelmann <sven.eckelmann@open-mesh.com>
Date: Sat, 9 Apr 2016 10:26:04 +0000
Subject: uboot-envtools: add OpenMesh OM5P-AC support

Signed-off-by: Sven Eckelmann <sven.eckelmann@open-mesh.com>

Backport of r49146

Forwarded: https://patchwork.ozlabs.org/patch/624196/

diff --git a/package/boot/uboot-envtools/files/ar71xx b/package/boot/uboot-envtools/files/ar71xx
index ef00f17..657b25f 100644
--- a/package/boot/uboot-envtools/files/ar71xx
+++ b/package/boot/uboot-envtools/files/ar71xx
@@ -28,6 +28,7 @@ mr900 | \
 mr900v2 | \
 nbg6716 | \
 om5p-an | \
+om5p-ac | \
 om5p | \
 tube2h | \
 wndr3700)
