From: Sven Eckelmann <sven.eckelmann@open-mesh.com>
Date: Sat, 9 Apr 2016 10:26:22 +0000
Subject: ar71xx: enable sysupgrade for the OpenMesh OM5P-ACv2

Signed-off-by: Sven Eckelmann <sven.eckelmann@open-mesh.com>

Backport of r49151

Forwarded: https://patchwork.ozlabs.org/patch/624201/

diff --git a/target/linux/ar71xx/base-files/lib/upgrade/openmesh.sh b/target/linux/ar71xx/base-files/lib/upgrade/openmesh.sh
index 1cfead9..209cdaa 100644
--- a/target/linux/ar71xx/base-files/lib/upgrade/openmesh.sh
+++ b/target/linux/ar71xx/base-files/lib/upgrade/openmesh.sh
@@ -74,6 +74,7 @@ platform_check_image_openmesh()
 			;;
 		OM5PAC)
 			[ "$board" = "om5p-ac" ] && break
+			[ "$board" = "om5p-acv2" ] && break
 			echo "Invalid image board target ($img_board_target) for this platform: $board. Use the correct image for this platform"
 			return 1
 			;;
diff --git a/target/linux/ar71xx/base-files/lib/upgrade/platform.sh b/target/linux/ar71xx/base-files/lib/upgrade/platform.sh
index 67c0661..4a4c476 100755
--- a/target/linux/ar71xx/base-files/lib/upgrade/platform.sh
+++ b/target/linux/ar71xx/base-files/lib/upgrade/platform.sh
@@ -308,7 +308,8 @@ platform_check_image() {
 	om2p-lc | \
 	om5p | \
 	om5p-an | \
-	om5p-ac)
+	om5p-ac | \
+	om5p-acv2)
 		platform_check_image_openmesh "$magic_long" "$1" && return 0
 		return 1
 		;;
@@ -542,7 +543,8 @@ platform_do_upgrade() {
 	om2p-lc | \
 	om5p | \
 	om5p-an | \
-	om5p-ac)
+	om5p-ac | \
+	om5p-acv2)
 		platform_do_upgrade_openmesh "$ARGV"
 		;;
 	unifi-outdoor-plus | \
