From: Sven Eckelmann <sven.eckelmann@open-mesh.com>
Date: Sat, 9 Apr 2016 10:26:34 +0000
Subject: ar71xx: add OM5P-ACv2 to the OM5P-AC profile

Signed-off-by: Sven Eckelmann <sven.eckelmann@open-mesh.com>

Backport of r49155

Forwarded: https://patchwork.ozlabs.org/patch/624205/

diff --git a/target/linux/ar71xx/generic/profiles/openmesh.mk b/target/linux/ar71xx/generic/profiles/openmesh.mk
index 64aaa24..c0919ed 100644
--- a/target/linux/ar71xx/generic/profiles/openmesh.mk
+++ b/target/linux/ar71xx/generic/profiles/openmesh.mk
@@ -28,12 +28,12 @@ endef
 $(eval $(call Profile,OM5P))
 
 define Profile/OM5PAC
-	NAME:=OpenMesh OM5P-AC
+	NAME:=OpenMesh OM5P-AC/OM5P-ACv2
 	PACKAGES:=kmod-ath9k kmod-ath10k om-watchdog ath10k-firmware-qca988x
 endef
 
 define Profile/OM5PAC/Description
-	Package set optimized for the OpenMesh OM5P-AC.
+	Package set optimized for the OpenMesh OM5P-AC/OM5P-ACv2.
 endef
 
 $(eval $(call Profile,OM5PAC))
