From: Jo-Philipp Wich <jo@mein.io>
Date: Wed, 30 Mar 2016 03:19:23 +0200
Subject: x86: search PARTUUID in any block device

Signed-off-by: Jo-Philipp Wich <jo@mein.io>

Backport of LEDE 924fb794bde41eca8289c6cf10322bf6683b4a80

diff --git a/target/linux/x86/base-files/lib/upgrade/platform.sh b/target/linux/x86/base-files/lib/upgrade/platform.sh
index f12deebf6484df6f3f69e453ad67688a76d57972..29eac77dfb0ae52f241696f3f62dce7d16106b20 100644
--- a/target/linux/x86/base-files/lib/upgrade/platform.sh
+++ b/target/linux/x86/base-files/lib/upgrade/platform.sh
@@ -17,7 +17,8 @@ platform_export_bootpart() {
 			PARTUUID=[a-f0-9][a-f0-9][a-f0-9][a-f0-9][a-f0-9][a-f0-9][a-f0-9][a-f0-9]-02)
 				uuid="${disk#PARTUUID=}"
 				uuid="${uuid%-02}"
-				for disk in /dev/[hsv]d[a-z] /dev/xvd[a-z]; do
+				for disk in /dev/*; do
+					[ -b "$disk" ] || continue
 					set -- $(dd if=$disk bs=1 skip=440 count=4 2>/dev/null | hexdump -v -e '4/1 "%02x "')
 					if [ "$4$3$2$1" = "$uuid" ]; then
 						export BOOTPART="${disk}1"
