From: Sven Eckelmann <sven.eckelmann@open-mesh.com>
Date: Fri, 20 May 2016 18:03:52 +0200
Subject: package/uboot-envtools: add OpenMesh OM2P-HSv3 support

Signed-off-by: Sven Eckelmann <sven.eckelmann@open-mesh.com>

Forwarded: https://patchwork.ozlabs.org/patch/637056/

diff --git a/package/boot/uboot-envtools/files/ar71xx b/package/boot/uboot-envtools/files/ar71xx
index 9071c11..81c6481 100644
--- a/package/boot/uboot-envtools/files/ar71xx
+++ b/package/boot/uboot-envtools/files/ar71xx
@@ -41,6 +41,7 @@ om2p | \
 om2pv2 | \
 om2p-hs | \
 om2p-hsv2 | \
+om2p-hsv3 | \
 om2p-lc)
 	ubootenv_add_uci_config "/dev/mtd1" "0x0" "0x40000" "0x40000"
 	;;
