From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Wed, 15 Jun 2016 19:31:08 +0200
Subject: scripts: feeds: fix version detection for Make >= 4.2.1

Signed-off-by: Jo-Philipp Wich <jo@mein.io>

Backport of LEDE df18b3756fef6a28d9194368d3156f0cd85fae19

diff --git a/scripts/feeds b/scripts/feeds
index f8f29cd..908e56c 100755
--- a/scripts/feeds
+++ b/scripts/feeds
@@ -22,7 +22,10 @@ my @mkver = split /\s+/, `$mk -v`, 4;
 my $valid_mk = 1;
 $mkver[0] =~ /^GNU/ or $valid_mk = 0;
 $mkver[1] =~ /^Make/ or $valid_mk = 0;
-$mkver[2] >= "3.81" or $valid_mk = 0;
+
+my ($mkv1, $mkv2) = split /\./, $mkver[2];
+($mkv1 >= 4 || ($mkv1 == 3 && $mkv2 >= 81)) or $valid_mk = 0;
+
 $valid_mk or die "Unsupported version of make found: $mk\n";
 
 my @feeds;
