From: Sven Eckelmann <sven.eckelmann@open-mesh.com>
Date: Fri, 20 May 2016 18:03:57 +0200
Subject: package/om-watchdog: add OpenMesh MR1750v2 support

Signed-off-by: Sven Eckelmann <sven.eckelmann@open-mesh.com>

Forwarded: https://patchwork.ozlabs.org/patch/637062/

diff --git a/package/kernel/om-watchdog/files/om-watchdog.init b/package/kernel/om-watchdog/files/om-watchdog.init
index bf8a124..8a1b66e 100644
--- a/package/kernel/om-watchdog/files/om-watchdog.init
+++ b/package/kernel/om-watchdog/files/om-watchdog.init
@@ -28,7 +28,7 @@ boot() {
 			"mr600v2")
 				service_start /sbin/om-watchdog 15
 				;;
-			"mr900"|"mr900v2"|"mr1750")
+			"mr900"|"mr900v2"|"mr1750"|"mr1750v2")
 				service_start /sbin/om-watchdog 16
 				;;
 		esac
