From: Ansuel Smith <ansuelsmth@gmail.com>
Date: Fri, 9 Mar 2018 02:29:58 +0100
Subject: e2fsprogs: bump to 1.44.0

Fix compilation error

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
(cherry picked from commit a9c00578b5c55357117db9dbbbd4e5652b9b4648)

diff --git a/tools/e2fsprogs/Makefile b/tools/e2fsprogs/Makefile
index 0a48b5434d91d57fb85061e2f0f088512d3cc692..d709e8fa88e80c75d8aa60dadeca9471821ba2e6 100644
--- a/tools/e2fsprogs/Makefile
+++ b/tools/e2fsprogs/Makefile
@@ -8,8 +8,8 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=e2fsprogs
-PKG_VERSION:=1.43.7
-PKG_HASH:=2a6367289047d68d9ba6a46cf89ab9a1efd0556cde02a51ebaf414ff51edded9
+PKG_VERSION:=1.44.0
+PKG_HASH:=a3567042f03fcee0e01db8f923b24cec92e2ba5b11f8e39e245672ad5392723d
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
