From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Sun, 7 Mar 2021 11:48:32 +0100
Subject: fastd: simplify Config.in

Signed-off-by: Matthias Schiffer <mschiffer@universe-factory.net>

diff --git a/net/fastd/Config.in b/net/fastd/Config.in
index 8302f7ee4dac874b1303ebeeb836551ef202c261..89ff6850aa5ab4ad0e762d8fb9473d5e5c820089 100644
--- a/net/fastd/Config.in
+++ b/net/fastd/Config.in
@@ -1,102 +1,79 @@
+if PACKAGE_fastd
+
 menu "Configuration"
-	depends on PACKAGE_fastd
 
 config FASTD_ENABLE_METHOD_CIPHER_TEST
 	bool "Enable cipher-test method provider"
-	depends on PACKAGE_fastd
-	default n
 
 config FASTD_ENABLE_METHOD_COMPOSED_GMAC
 	bool "Enable composed-gmac method provider"
-	depends on PACKAGE_fastd
+	select FASTD_ENABLE_MAC_GHASH
 	default y
 
 config FASTD_ENABLE_METHOD_COMPOSED_UMAC
 	bool "Enable composed-umac method provider"
-	depends on PACKAGE_fastd
+	select FASTD_ENABLE_MAC_UHASH
 	default y
 
 config FASTD_ENABLE_METHOD_GENERIC_GMAC
 	bool "Enable generic-gmac method provider"
-	depends on PACKAGE_fastd
+	select FASTD_ENABLE_MAC_GHASH
 	default y
 
 config FASTD_ENABLE_METHOD_GENERIC_POLY1305
 	bool "Enable generic-poly1305 method provider"
-	depends on PACKAGE_fastd
-	default n
 
 config FASTD_ENABLE_METHOD_GENERIC_UMAC
 	bool "Enable generic-umac method provider"
-	depends on PACKAGE_fastd
+	select FASTD_ENABLE_MAC_UHASH
 	default y
 
 config FASTD_ENABLE_METHOD_NULL
 	bool "Enable null method"
-	depends on PACKAGE_fastd
 	default y
 
 
 config FASTD_ENABLE_CIPHER_NULL
 	bool "Enable the null cipher"
-	depends on PACKAGE_fastd
 	default y
 
 config FASTD_ENABLE_CIPHER_SALSA20
 	bool "Enable the Salsa20 cipher"
-	depends on PACKAGE_fastd
-	default n
 
 config FASTD_ENABLE_CIPHER_SALSA2012
 	bool "Enable the Salsa20/12 cipher"
-	depends on PACKAGE_fastd
 	default y
 
 
 config FASTD_ENABLE_MAC_GHASH
-	bool "Enable the GHASH message authentication code"
-	depends on PACKAGE_fastd
-	default y
+	bool
 
 config FASTD_ENABLE_MAC_UHASH
-	bool "Enable the UHASH message authentication code"
-	depends on PACKAGE_fastd
-	default y
+	bool
 
 
 config FASTD_WITH_CAPABILITIES
 	bool "Enable POSIX capability support"
-	depends on PACKAGE_fastd
-	default n
 
 config FASTD_WITH_CMDLINE_USER
 	bool "Include support for setting user/group related options on the command line"
-	depends on PACKAGE_fastd
-	default n
 
 config FASTD_WITH_CMDLINE_LOGGING
 	bool "Include support for setting logging related options on the command line"
-	depends on PACKAGE_fastd
-	default n
 
 config FASTD_WITH_CMDLINE_OPERATION
 	bool "Include support for setting options related to the VPN operation (like mode, interface, encryption method) on the command line"
-	depends on PACKAGE_fastd
-	default n
 
 config FASTD_WITH_CMDLINE_COMMANDS
 	bool "Include support for setting handler scripts (e.g. --on-up) on the command line"
-	depends on PACKAGE_fastd
-	default n
 
 config FASTD_WITH_DYNAMIC_PEERS
 	bool "Include support for dynamic peers (using on-verify handlers)"
-	depends on PACKAGE_fastd
-	default n
 
 config FASTD_WITH_STATUS_SOCKET
 	bool "Include support for status sockets"
-	depends on PACKAGE_fastd
 	default y
 
 endmenu
+
+endif
