From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Thu, 13 Jul 2017 17:15:21 +0200
Subject: procd: switch to latest master

diff --git a/package/system/procd/Makefile b/package/system/procd/Makefile
index c260a371ac951a5efa125c2ce5bbfcf8d5d2779b..6c690e49f4a0312a8a6eed13ab36b8a67a485556 100644
--- a/package/system/procd/Makefile
+++ b/package/system/procd/Makefile
@@ -12,9 +12,9 @@ PKG_RELEASE:=1
 
 PKG_SOURCE_PROTO:=git
 PKG_SOURCE_URL=$(LEDE_GIT)/project/procd.git
-PKG_SOURCE_DATE:=2018-01-04
-PKG_SOURCE_VERSION:=188353099cf6fc88f145cfcb84a4db3f6523528a
-PKG_MIRROR_HASH:=52605b2a3caf59cdbf88ba218c6288cbe075749422efd69e985ae75bb1001488
+PKG_SOURCE_DATE:=2018-12-02
+PKG_SOURCE_VERSION:=a5954cf302fe46d8e29770c818336763112b9a6e
+PKG_MIRROR_HASH:=1afa1f0160dd8ffac310e75519601c1f9a6107d818f405bb3fe9ef3e7143c2e3
 CMAKE_INSTALL:=1
 
 PKG_LICENSE:=GPL-2.0
