From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Sat, 8 Jul 2017 14:44:00 +0200
Subject: procd: switch to master branch

diff --git a/package/system/procd/Makefile b/package/system/procd/Makefile
index 70515958cd5002841fd9698b7a7f4aec01ddfb3b..cead3b9b7a5f83879a1aee08ab0463049dd228f4 100644
--- a/package/system/procd/Makefile
+++ b/package/system/procd/Makefile
@@ -13,8 +13,8 @@ PKG_RELEASE:=1
 PKG_SOURCE_PROTO:=git
 PKG_SOURCE_URL=$(LEDE_GIT)/project/procd.git
 PKG_SOURCE_DATE:=2017-06-19
-PKG_SOURCE_VERSION:=89918c8cb78e651ee0922f13a96b7f25fc2f285a
-PKG_MIRROR_HASH:=fad1a37227f85070f6a49a37ea1e98c22c066c91f16ad26898152972eb0c6a91
+PKG_SOURCE_VERSION:=453116e08e6a9349374bbff427b75f57ce5387c9
+PKG_MIRROR_HASH:=a0d5c1ddbd2ba63d52125fa5a815d48a3bfb1b1e0a6b85f3335af8a873ed58fc
 CMAKE_INSTALL:=1
 
 PKG_LICENSE:=GPL-2.0
