From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Thu, 13 Jul 2017 17:15:21 +0200
Subject: procd: switch to latest master

diff --git a/package/system/procd/Makefile b/package/system/procd/Makefile
index 864d3f62a9c3e0f9ab1b72b1ede8205a2edd83d4..ea014b5b3461e1ed55fb272592a1c2be6535c2c8 100644
--- a/package/system/procd/Makefile
+++ b/package/system/procd/Makefile
@@ -12,9 +12,9 @@ PKG_RELEASE:=1
 
 PKG_SOURCE_PROTO:=git
 PKG_SOURCE_URL=$(LEDE_GIT)/project/procd.git
-PKG_SOURCE_DATE:=2017-11-14
-PKG_SOURCE_VERSION:=d9dc0e03d70937dbbea12af86997701fbd717dc5
-PKG_MIRROR_HASH:=ad62410e6e43113c13aebf15f60c145f1eac09f309b8e7bd006a3621ca8f07a5
+PKG_SOURCE_DATE:=2017-10-10
+PKG_SOURCE_VERSION:=7c9aa7c39362bca872905e906ac82dfc667a4112
+PKG_MIRROR_HASH:=85ee6d61e3e429f033e7d192a1f3ae749cf863e46650ab4a093cf578074054de
 CMAKE_INSTALL:=1
 
 PKG_LICENSE:=GPL-2.0
