From: Piotr Dymacz <pepe2k@gmail.com>
Date: Tue, 17 Oct 2017 23:32:11 +0200
Subject: uboot-envtools: add support for GL.iNet GL-AR750

Signed-off-by: Piotr Dymacz <pepe2k@gmail.com>

diff --git a/package/boot/uboot-envtools/files/ar71xx b/package/boot/uboot-envtools/files/ar71xx
index 25bec7eb33fd691bae417512c4c03e133d93213d..26f1ff938b8191e84315cf308f78e0907264c671 100644
--- a/package/boot/uboot-envtools/files/ar71xx
+++ b/package/boot/uboot-envtools/files/ar71xx
@@ -29,6 +29,7 @@ cr3000|\
 cr5000|\
 eap300v2|\
 gl-ar300m|\
+gl-ar750|\
 hornet-ub|\
 hornet-ub-x2|\
 jwap230|\
