From: Henryk Heisig <hyniu@o2.pl>
Date: Thu, 29 Jun 2017 15:38:22 +0200
Subject: ar71xx: fix board.bin used by QCA9886 in Archer C58/C59/C60

Signed-off-by: Henryk Heisig <hyniu@o2.pl>
(cherry picked from commit e917e51bf91fc7cb5085bda5e67d62520801f9cc)

diff --git a/target/linux/ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata b/target/linux/ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
index 538c86e4743109f7665096da32620d7862248aea..cfdc20455d61c3900473f57c3267b1a3fd10e150 100644
--- a/target/linux/ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
+++ b/target/linux/ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
@@ -138,6 +138,8 @@ case "$FIRMWARE" in
 	case $board in
 	archer-c58-v1)
 		ath10kcal_extract "art" 20480 12064
+		ln -sf /lib/firmware/ath10k/pre-cal-pci-0000\:00\:00.0.bin \
+			/lib/firmware/ath10k/QCA9888/hw2.0/board.bin
 		;;
 	esac
 	;;
