From: David Bauer <mail@david-bauer.net>
Date: Thu, 11 Jan 2018 21:23:30 +0100
Subject: ar71xx: omit VLAN from Archer C58/C59 LAN interface

diff --git a/target/linux/ar71xx/base-files/etc/board.d/02_network b/target/linux/ar71xx/base-files/etc/board.d/02_network
index 57bc912aa5ede22a3fad4af6a1deb00c33f17cf8..1690172b5dfac1e1aa278c38c27ba64cfd61ade0 100755
--- a/target/linux/ar71xx/base-files/etc/board.d/02_network
+++ b/target/linux/ar71xx/base-files/etc/board.d/02_network
@@ -208,7 +208,7 @@ ar71xx_setup_interfaces()
 	archer-c58-v1|\
 	archer-c59-v1|\
 	rb-450g)
-		ucidef_set_interfaces_lan_wan "eth1.1" "eth0"
+		ucidef_set_interfaces_lan_wan "eth1" "eth0"
 		ucidef_add_switch "switch0" \
 			"0@eth1" "1:lan:1" "2:lan:4" "3:lan:3" "4:lan:2"
 		;;
