From: Alexander Couzens <lynxis@fe80.eu>
Date: Sun, 11 Jun 2017 12:43:24 +0200
Subject: ar71xx/ag71xx_ar7240_get_port_link: fix off-by-one check on argument `port`

Found-by: Coverity Scan #1329901
Signed-off-by: Alexander Couzens <lynxis@fe80.eu>

diff --git a/target/linux/ar71xx/files/drivers/net/ethernet/atheros/ag71xx/ag71xx_ar7240.c b/target/linux/ar71xx/files/drivers/net/ethernet/atheros/ag71xx/ag71xx_ar7240.c
index c5aed0d2dbfcc726f54ca7eaf7dde2f7aba03ca7..e457acb50cf03e2ece5584843d6ce437da8bb8cb 100644
--- a/target/linux/ar71xx/files/drivers/net/ethernet/atheros/ag71xx/ag71xx_ar7240.c
+++ b/target/linux/ar71xx/files/drivers/net/ethernet/atheros/ag71xx/ag71xx_ar7240.c
@@ -963,7 +963,7 @@ ar7240_get_port_link(struct switch_dev *dev, int port,
 	struct mii_bus *mii = as->mii_bus;
 	u32 status;
 
-	if (port > AR7240_NUM_PORTS)
+	if (port >= AR7240_NUM_PORTS)
 		return -EINVAL;
 
 	status = ar7240sw_reg_read(mii, AR7240_REG_PORT_STATUS(port));
@@ -1000,7 +1000,7 @@ ar7240_get_port_stats(struct switch_dev *dev, int port,
 {
 	struct ar7240sw *as = sw_to_ar7240(dev);
 
-	if (port > AR7240_NUM_PORTS)
+	if (port >= AR7240_NUM_PORTS)
 		return -EINVAL;
 
 	ar7240sw_capture_stats(as);
