From: Felix Fietkau <nbd@nbd.name>
Date: Wed, 28 Feb 2018 13:57:58 +0100
Subject: ar71xx: fix build with ag71xx debugfs support enabled

Signed-off-by: Felix Fietkau <nbd@nbd.name>

diff --git a/target/linux/ar71xx/files/drivers/net/ethernet/atheros/ag71xx/ag71xx_debugfs.c b/target/linux/ar71xx/files/drivers/net/ethernet/atheros/ag71xx/ag71xx_debugfs.c
index 89cea0c0f5ae97c9917cd13e1ee89bd31fe45e58..20910b9681a54e53f3e089d0a589e1b2b45c1833 100644
--- a/target/linux/ar71xx/files/drivers/net/ethernet/atheros/ag71xx/ag71xx_debugfs.c
+++ b/target/linux/ar71xx/files/drivers/net/ethernet/atheros/ag71xx/ag71xx_debugfs.c
@@ -175,8 +175,8 @@ static ssize_t read_file_ring(struct file *file, char __user *user_buf,
 		return -ENOMEM;
 
 	len += snprintf(buf + len, buflen - len,
-			"Idx ... %-8s %-8s %-8s %-8s . %-10s\n",
-			"desc", "next", "data", "ctrl", "timestamp");
+			"Idx ... %-8s %-8s %-8s %-8s .\n",
+			"desc", "next", "data", "ctrl");
 
 	spin_lock_irqsave(&ag->lock, flags);
 
@@ -184,12 +184,11 @@ static ssize_t read_file_ring(struct file *file, char __user *user_buf,
 	dirty = (ring->dirty & ring_mask);
 	desc_hw = ag71xx_rr(ag, desc_reg);
 	for (i = 0; i < ring_size; i++) {
-		struct ag71xx_buf *ab = &ring->buf[i];
 		struct ag71xx_desc *desc = ag71xx_ring_desc(ring, i);
 		u32 desc_dma = ((u32) ring->descs_dma) + i * AG71XX_DESC_SIZE;
 
 		len += snprintf(buf + len, buflen - len,
-			"%3d %c%c%c %08x %08x %08x %08x %c %10lu\n",
+			"%3d %c%c%c %08x %08x %08x %08x %c\n",
 			i,
 			(i == curr) ? 'C' : ' ',
 			(i == dirty) ? 'D' : ' ',
@@ -198,8 +197,7 @@ static ssize_t read_file_ring(struct file *file, char __user *user_buf,
 			desc->next,
 			desc->data,
 			desc->ctrl,
-			(desc->ctrl & DESC_EMPTY) ? 'E' : '*',
-			ab->timestamp);
+			(desc->ctrl & DESC_EMPTY) ? 'E' : '*');
 	}
 
 	spin_unlock_irqrestore(&ag->lock, flags);
